streams/src/main/java/org/apache/kafka/streams/processor/internals/ClientUtils.java

```diff
@@ -90,24 +96,21 @@ public static String getTaskProducerClientId(final String threadClientId, final
         return result;
     }
 
-    public static Map<TopicPartition, ListOffsetsResultInfo> fetchEndOffsetsWithoutTimeout(final Collection<TopicPartition> partitions,
-                                                                                           final Admin adminClient) {
-        return fetchEndOffsets(partitions, adminClient, null);
+    public static int getAdminDefaultApiTimeoutMs(final StreamsConfig streamsConfig) {
+        final InternalAdminClientConfig dummyAdmin = new InternalAdminClientConfig(streamsConfig.getAdminConfigs("dummy"));
+        return dummyAdmin.getInt(AdminClientConfig.DEFAULT_API_TIMEOUT_MS_CONFIG);
     }
 
     public static Map<TopicPartition, ListOffsetsResultInfo> fetchEndOffsets(final Collection<TopicPartition> partitions,
                                                                              final Admin adminClient,
-                                                                             final Duration timeout) {
+                                                                             final long timeoutMs) {
```

Comment:
It seems to me ```Duration``` is more readable than ```long```. Is there a reason to make this change?

Reply:
I think it's because we're now also calling it right after calling `getAdminDefaultApiTimeoutMs`, so it seems a bummer to create a Duration from millis and then immediately convert it back to millis.

Reply:
Yep